File: smartmoneyserver/src/main/java/fpt/aptech/server/enums/notification/NotificationType.java
package fpt.aptech.server.enums.notification;

import lombok.Getter;

/**
 * Định nghĩa các loại thông báo trong hệ thống.
 * Dùng cho tNotifications.notify_type
 */
@Getter
public enum NotificationType {
    TRANSACTION(1),     // Giao dịch / Biến động số dư
    SAVING(2),          // Mục tiêu tiết kiệm / Quỹ
    BUDGET(3),          // Cảnh báo ngân sách / Vượt hạn mức
    SYSTEM(4),          // Hệ thống / Cập nhật / Bảo mật
    CHAT_AI(5),         // Thông báo từ trợ lý AI
    WALLETS(6),         // Thông báo liên quan đến ví / số dư âm
    EVENTS(7),          // Sự kiện / Lịch trình
    DEBT_LOAN(8),       // Nhắc nợ / Thu nợ
    REMINDER(9);        // Nhắc nhở chung / Daily nhắc ghi chép

    private final int value;

    NotificationType(int value) {
        this.value = value;
    }
}

File: APP MOBILE/smart_money/lib/modules/notification/enums/notification_type.dart
/// Loại thông báo: Giao dịch, Tiết kiệm, Ngân sách, Hệ thống, Chat AI, Ví, Sự kiện, Nợ/Vay, Nhắc nhở
enum NotificationType {
  transaction,  // 1 - Giao dịch / Biến động số dư
  saving,       // 2 - Mục tiêu tiết kiệm / Quỹ
  budget,       // 3 - Cảnh báo ngân sách
  system,       // 4 - Hệ thống / Cập nhật
  chatAi,       // 5 - Thông báo từ AI
  wallets,      // 6 - Thông báo ví / số dư âm
  events,       // 7 - Sự kiện / Lịch trình
  debtLoan,     // 8 - Nhắc nợ / Thu nợ
  reminder;     // 9 - Nhắc nhở chung

  int get value => index + 1;

  static NotificationType fromValue(int value) {
    if (value < 1 || value > 9) {
      return NotificationType.system;
    }
    return NotificationType.values[value - 1];
  }

  String get displayName {
    switch (this) {
      case NotificationType.transaction:
        return 'Transaction';
      case NotificationType.saving:
        return 'Savings';
      case NotificationType.budget:
        return 'Budget';
      case NotificationType.system:
        return 'System';
      case NotificationType.chatAi:
        return 'Chat AI';
      case NotificationType.wallets:
        return 'Wallet';
      case NotificationType.events:
        return 'Event';
      case NotificationType.debtLoan:
        return 'Debt/Loan';
      case NotificationType.reminder:
        return 'Reminder';
    }
  }
}

